sys/linux/dev_bifrost.txt

```diff
@@ -0,0 +1,370 @@
+# Copyright 2020 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <asm/page.h>
+include <drivers/gpu/arm/bifrost/mali_kbase_debug.h>
+include <drivers/gpu/arm/bifrost/mali_kbase_hwcnt_reader.h>
+include <drivers/gpu/arm/bifrost/mali_kbase_ioctl.h>
+include <drivers/gpu/arm/bifrost/mali_base_kernel.h>
+include <uapi/linux/fcntl.h>
+
+resource fd_bifrost[fd]
+resource gpu_addr[int64]: BASEP_MEM_INVALID_HANDLE, BASEP_MEM_WRITE_ALLOC_PAGES_HANDLE
+resource user_addr[int64]: -1
+resource fd_fence[fd]
+resource fd_hwcnt[fd]
+
+openat$bifrost(fd const[AT_FDCWD], file ptr[in, string["/dev/bifrost"]], flags flags[open_flags], mode const[0]) fd_bifrost
+
+mmap$bifrost(addr vma, len len[addr], prot flags[mmap_prot], flags flags[mmap_flags], fd fd_bifrost, offset fileoff) user_addr
```

Comment:
As far as I remember the driver has a check for NULL address, if so it will be better to use `const[0]` for the addr argument.

Reply:
Done.